frontend/src/Commons/Forms/InputModules/FieldsAndSoil/FieldsAndSoil.tsx

```diff
@@ -17,117 +17,140 @@ import CustomTextArea from '@Commons/Input/TextArea/CustomTextArea';
 import { faWheatAwn } from '@fortawesome/free-solid-svg-icons';
 import initialFarmDetails from '@Constants/InitialFarmDetails';
 import FieldDetailInterface from 'src/Interface/FieldDetailsInterface';
-import {
-  StyledFarmInfo,
-  StyledTextAreaContainer,
-  StyledButtonGroupContainer,
-  StyledListContainer,
-  StyledListItem,
-} from './FieldsAndSoil.style';
-
-const initialFieldValues = initialFarmDetails.Fields[0];
+import FieldsButtonComponent from './FieldsButtonComponent';
+import FieldsListComponent from './FieldsListComponent';
+import { StyledFarmInfo, StyledTextAreaContainer } from './FieldsAndSoil.style';
+import { StyledButtonGroupContainer } from './FieldsButtonComponent.styles';
 
 const FieldsAndSoilComponent: React.FC<InputModuleProps> = ({ farmDetails, updateFarmDetails }) => {
-  const [fieldsInfo, setFieldsInfo] = useState(farmDetails);
+  // Builds field info inside the field form module.
+  const [, setFieldsInfo] = useState(farmDetails);
+  const [fieldIndex, setFieldIndex] = useState(0);
+  const [initialFieldValues, setInitialFieldValues] = useState(
+    initialFarmDetails.Fields[fieldIndex],
+  );
+  // Only triggered once, it would show list and persists.
+  const [isSubmitted, setSubmitted] = useState<boolean>(false);
+  // Would trigger when new field button is clicked.
+  const [fieldAdd, setFieldAdd] = useState<boolean>(false);
```

Comment:
'fieldAdd' sounds like an action, could we change this to 'fieldAdded' like you did for 'isSubmitted' so it's easier to know this is a state? 

I'm only commenting about this because I found this check a bit confusing:

```typescript
        {!fieldAdd && (
            <FieldsButtonComponent
              addNewField={addNewField}
              submitFarmInfo={submitFarmInfo}
            />
```

Reply:
For consistency

Reply:
Couldn't this condition just check if the length of fields is greater than 0?

Reply:
I have to use the state for this. using farmdetails.Fields.length will be on as long as the length is greater than 0. I need the addNewField, back, and next to be not displayed when user is using the input fields, because they will just press next and there would be some confusion. I need those buttons displaying when the forms are not displaying.
For reference:
![image](https://github.com/bcgov/nr-sustainment-capstone-2024/assets/144957476/a8e4c09a-9931-4522-a918-c0cc812860c8)